migration/v0.43/apply/20-upgrade-opensearch.sh

```diff
@@ -0,0 +1,155 @@
+            log_info "- Successfully disabled read-only mode for '${os_dashboards_index}'"
+          else
+            log_fatal "Failed to disable read-only mode for '${os_dashboards_index}'" "${resp}"
+          fi
+
+          log_info "- Disabling read-only mode for .kibana_1"
+          resp=$(curl -sS -kL -u "${user}:${password}" -X PUT "${os_url}"/.kibana_1/_settings -H 'Content-Type: application/json' -d'
+          {
+            "settings": {
+              "index.blocks.write": false
+            }
+          }
+          ')
+          acknowledged=$(echo "${resp}" | grep "^{" | jq -r '.acknowledged')
+          if [ "${acknowledged}" = "true" ]; then
+            log_info "- Successfully disabled read-only mode for .kibana_1"
+          else
+            log_fatal "Failed to disable read-only mode for .kibana_1" "${resp}"
+          fi
+        else
+          log_fatal "Failed to check if index .kibana_1 already exists" "${resp}"
+        fi
+
+        resp=$(curl -sS -kL -u "${user}:${password}" -X GET "${os_url}"/_alias/.kibana)
+        if [[ $(echo "${resp}" | jq -r 'to_entries | .[0].key') == .kibana* ]]; then
+          log_info "- Alias .kibana already exists, skipping"
+        elif [[ $(echo "${resp}" | jq -r '.status') == "404" ]]; then
+          log_info "- Creating alias .kibana"
+          resp=$(curl -sS -kL -u "${user}:${password}" -X PUT "${os_url}"/.kibana_1/_aliases/.kibana)
+          acknowledged=$(echo "${resp}" | grep "^{" | jq -r '.acknowledged')
+          if [ "${acknowledged}" = "true" ]; then
+            log_info "- Successfully created alias .kibana"
+          else
+            log_fatal "Failed to create alias .kibana" "${resp}"
+          fi
+        else
+          log_fatal "Failed to check if alias .kibana exists" "${resp}"
+        fi
+
+        os_dashboards_index=$(curl -sS -kL -u "${user}:${password}" -X GET "${os_url}"/_alias/.opensearch_dashboards | jq -r 'to_entries | .[0].key')
+        if [[ "${os_dashboards_index}" != .opensearch_dashboards* ]]; then
+          log_info "- Skipping: Alias .opensearch_dashboards doesn't exist"
+        else
+          log_info "- Deleting index '${os_dashboards_index}'"
+          resp=$(curl -sS -kL -u "${user}:${password}" -X DELETE "${os_url}"/"${os_dashboards_index}")
+          acknowledged=$(echo "${resp}" | grep "^{" | jq -r '.acknowledged')
+          if [ "${acknowledged}" = "true" ]; then
+            log_info "- Successfully deleted index '${os_dashboards_index}'"
+          else
+            log_fatal "Failed to delete index '${os_dashboards_index}'" "${resp}"
+          fi
+        fi
+
+        log_info "- Removing opensearch-configurer"
+        helmfile_destroy sc name=opensearch-configurer
+        log_info "- Upgrading Opensearch"
+        helmfile_do sc -lapp=opensearch sync
+
+      else
+
+        log_info "- OpenSearch chart version already 2.26.1, executing generic apply"
+        helmfile_do sc -lapp=opensearch apply
```

Comment:
```suggestion
        helmfile_upgrade sc app=opensearch
```
Prefer to use this when you aren't syncing as it should be a bit safer if something goes wrong.